### PR TITLE
Set proper file timestamp when copying from unzipped folder

### PR DESCRIPTION
**Pull request by alpaix**
Fixes https://github.com/NuGet/Home/issues/420

### CODE REVIEW

**Comment by alpaix** [comment]
//cc @yishaigalatzer @emgarten @deepakaravindr

**Comment by forderud** [comment]
I've now built this branch locally, and hereby confirm that this pull request fixes https://github.com/NuGet/Home/issues/420 for me.

I am looking forward for this pull request to be merged. Thanks a lot for your effort on fixing this problem @alpaix!

File: src/NuGet.Core/NuGet.Packaging/PackageReaderBase.cs [hunk 1]
```diff
@@ -52,6 +55,88 @@ public PackageReaderBase(IFrameworkNameProvider frameworkProvider, IFrameworkCom
...
+                if (_nuspecReader == null)
+                {
+                    _nuspecReader = new NuspecReader(GetNuspec());
+                }
+
+                return _nuspecReader;
+            }
+        }
+
+        protected virtual void Dispose(bool disposing)
```
Comment (emgarten) [inline src/NuGet.Core/NuGet.Packaging/PackageReaderBase.cs]: Should this call Dispose() ?

File: src/NuGet.Core/NuGet.Packaging/PackageReaderExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
+﻿using NuGet.Packaging.Core;
+using System;
+using System.Collections.Generic;
+using System.IO;
```
Comment (emgarten) [inline src/NuGet.Core/NuGet.Packaging/PackageReaderExtensions.cs]: Update your visual studio settings to put system first

File: test/NuGet.Core.Tests/NuGet.PackageManagement.Test/NuGetPackageManagerTests.cs [hunk 3]
```diff
@@ -20,6 +20,7 @@
 using NuGet.Versioning;
 using Test.Utility;
 using Xunit;
+using Moq;
```
Comment (emgarten) [inline test/NuGet.Core.Tests/NuGet.PackageManagement.Test/NuGetPackageManagerTests.cs]: Sort

**Comment by emgarten** [comment]
Looks good to me :shipit:

**Pull request merged**